library/core/src/result.rs

```diff
@@ -853,6 +853,35 @@ impl<T, E> Result<T, E> {
             Err(e) => Err(op(e)),
         }
     }
+    
+    /// Returns `Err(e)` if the result is `Err(e)` (where `e` is the wrapped error),
+    /// otherwise calls `predicate` with the wrapped value and returns:
```

Comment:
in libstd a "predicate" is almost *always* a function returning `bool`.

places where the word "predicate" is used:

* {BinaryHeap, BTreeSet, ...}::retain
* {BTreeMap, Vec, ...}::drain_filter
* VecDeque::partition_point
* Option::filter
* Iterator::{filter, skip_while, take_while, rfind, partition, partition_in_place, is_partitioned, all, any, find}
* [T]::{split, split_inclusive, rsplit, splitn, rsplitn, splitn_mut, rsplitn_mut, group_by, group_by_mut, partition_point}
* CondVar::{wait, wait_timeout}

The only exception to `* -> bool` is `Iterator::map_while` takes is a `T -> Option<U>`, but technically this input is a "predicate and map" by combining `map`+`take_while`.


in the other words, the input of this `check` function shouldn't be called a predicate.

Reply:
`Option::filter` calls it a predicate.

Reply:
where you take inspiration doesn't matter, the truth is a `* -> Result<(), E>` has never been called a predicate.